google-datacatalog-sisense-connector/src/google/datacatalog_connectors/sisense/scrape/rest_api_helper.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/python
+#
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import requests
+from typing import Dict, List
+
+from google.datacatalog_connectors.sisense.scrape import \
+    authenticator, constants
+
+
+class RESTAPIHelper:
+
+    def __init__(self, server_address: str, api_version: str, username: str,
+                 password: str):
+
+        self.__server_address = server_address
+        self.__api_version = api_version
+        self.__username = username
+        self.__password = password
+
+        self.__base_api_endpoint = f'{server_address}/api/{api_version}'
+        self.__common_headers = {
+            'Content-Type': constants.JSON_CONTENT_TYPE,
+            'Accept': constants.JSON_CONTENT_TYPE
+        }
+
+        self.__auth_credentials = None
+
+    def get_all_folders(self) -> List[Dict]:
+        """Get all Folders the user has access to on a given server.
+
+        Returns:
+            A ``list``.
+        """
+        self.__set_up_auth()
+
+        # Sisense can return the folders in flat (default) or tree structures.
+        # We decided for flat because it simplifies further processing.
+        url = f'{self.__base_api_endpoint}/folders?structure=flat'
```

Comment:
I saw that Sisense API's support pagination using parameters like limit and skip(offset), as reference: https://sisense.dev/guides/rest/using-rest-api.html#limit-skip

What about creating a method which handles pagination, which will be used for all API calls?
Maybe folder is a level that wouldn't need pagination.... don't see having like thousands of folders... but once you get to the field and widget level pagination will be important.

WDYT? If you want to do it in a next PR let me know.

Reply:
Nice catch, I like the idea!
I'm going to merge the present PR but keep this conversation unresolved until I create the next one with the pagination handling.
Thanks! 😃

Reply:
Pagination logic handled in this PR: https://github.com/GoogleCloudPlatform/datacatalog-connectors-bi/pull/75.